turnstile-example/turnstile/examples/cs478/2021-02-18/stlc.rkt

```diff
@@ -62,6 +66,12 @@
    ---------------------
    [⊢ (λ- (x-) e-)]])
 
+#;(define-typerule (+ e1 e2) ≫
```

Comment:
I commented this out b/c the `#%app` rule was screwing with it, I'm assuming it would work if I created two argument applications.

Reply:
It should still work, even with `#%app`. Because it's essentially just adding a new kind of term to the "t" grammar in the book.

In this particular case, it fails because defining `#%app` in this file *overloads all fn applications to be typed fn application*. So the call to untyped `+-` below will crash. You'll need to insert an explicit `#%app-`, like you did with `succ` and friends.

But this does become redundant once we add `#%app`. Instead of manually specifying how to type check `+` (and every other operator) we should probably just to use `define-primop` to assign `+` a type, and let `#%app` handle the rest of type checking.